google/resource_endpoints_service.go

```diff
@@ -0,0 +1,297 @@
+package google
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/servicemanagement/v1"
+)
+
+func resourceEndpointsService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEndpointsServiceCreate,
+		Read:   resourceEndpointsServiceRead,
+		Delete: resourceEndpointsServiceDelete,
+		Update: resourceEndpointsServiceUpdate,
+		Schema: map[string]*schema.Schema{
+			"service_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"openapi_config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"grpc_config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"protoc_output": &schema.Schema{
```

Comment:
I am not sure I like the name protoc_output, seems like a computed field to me at first glance. Would `protoc_config` be a better name?

Reply:
Well, it's not a config - it's just a giant raw binary file which is output by protoc.  I can rename it to, maybe, `protoc_service_descriptor_bytes` or something like that - thoughts?